docs/examples.rst

```diff
@@ -0,0 +1,307 @@
+.. _ksql_examples:
+
+KSQL Examples
+=============
+
+These examples use a ``pageviews`` stream and a ``users`` table.
+
+.. contents:: Contents
+    :local:
+    :depth: 1
+
+
+Creating streams
+----------------
+
+Prerequisite:
+    The corresponding Kafka topics must already exist in your Kafka cluster.
+
+Create a stream with three columns on the Kafka topic that is named ``pageviews``. It is important to instruct KSQL the format
+of the values that are stored in the topic. In this example, the values format is ``DELIMITED``.
+
+.. code:: sql
+
+    CREATE STREAM pageviews \
+      (viewtime BIGINT, \
+       userid VARCHAR, \
+       pageid VARCHAR) \
+      WITH (kafka_topic='pageviews-topic', \
+            value_format='DELIMITED');
+
+**Associating Kafka message keys:** The above statement does not make
+any assumptions about the Kafka message key in the underlying Kafka
+topic. However, if the value of the message key in Kafka is the same as
+one of the columns defined in the stream in KSQL, you can provide such
+information in the WITH clause. For instance, if the Kafka message key
+has the same value as the ``pageid`` column, you can write the CREATE
+STREAM statement as follows:
+
+.. code:: sql
+
+    CREATE STREAM pageviews \
+      (viewtime BIGINT, \
+       userid VARCHAR, \
+       pageid VARCHAR) \
+      WITH (kafka_topic='pageviews-topic', \
+            value_format='DELIMITED', \
+            key='pageid');
+
+**Associating Kafka message timestamps:** If you want to use the value
+of one of the columns as the Kafka message timestamp, you can provide
+such information to KSQL in the WITH clause. The message timestamp is
+used in window-based operations in KSQL (such as windowed aggregations)
+and to support event-time based processing in KSQL. For instance, if you
+want to use the value of the ``viewtime`` column as the message
+timestamp, you can rewrite the above statement as follows:
+
+.. code:: sql
+
+    CREATE STREAM pageviews \
+      (viewtime BIGINT, \
+       userid VARCHAR, \
+       pageid VARCHAR) \
+      WITH (kafka_topic='pageviews-topic', \
+            value_format='DELIMITED', \
+            key='pageid', \
+            timestamp='viewtime');
+
+Creating tables
+---------------
+
+Prerequisite:
+    The corresponding Kafka topics must already exist in your Kafka cluster.
+
+Create a table with several columns. In this example, the table has columns with primitive data
+types, a column of ``array`` type, and a column of ``map`` type:
+
+.. code:: sql
+
+    CREATE TABLE users \
+      (registertime BIGINT, \
+       gender VARCHAR, \
+       regionid VARCHAR, \
+       userid VARCHAR, \
+       interests array<VARCHAR>, \
+       contact_info map<VARCHAR, VARCHAR>) \
+      WITH (kafka_topic='users-topic', \
+            value_format='JSON');
```

Comment:
please add `key = 'userid'` to the `WITH` section similar to the create stream example.